lib/file/data.dart

```diff
@@ -68,6 +71,24 @@ class File implements Entity<File>, Comparable<File> {
 
   @HiveField(1)
   final String name;
+  String get extension {
+    final lastDot = name.lastIndexOf('.');
+    return lastDot == null ? null : name.substring(lastDot + 1);
+  }
+
+  Future<io.File> get localFile async {
+    final directory = await getExternalStorageDirectory();
+    final fileName = extension == null ? id.toString() : '$id.$extension';
+    return io.File('${directory.path}/$fileName');
+  }
```

Comment:
You might want to simplify this code with [`path`](https://pub.dev/packages/path)